src/test/TUSDCurveCallee.t.sol

```diff
@@ -0,0 +1,383 @@
+        // Enable liquidations
+        Clipper(clipper).file("stopped", 0);
+
+        // Use Abacus/LinearDecrease – tau [seconds]
+        Clipper(clipper).file("calc", address(calc));
+
+        // Set liquidation penalty (chop) to 0
+        Dog(dog).file("TUSD-A", "chop", 1.0 * WAD);
+
+        // Set LR to 150%
+        // This is simulated in newAuction() after opening a new auction
+
+        // Set buff to 1
+        Clipper(clipper).file("buf", 1.0 * RAY); // 0% Initial price buffer
+
+        // Set tau to 21,600,000 second (est. 10bps drop per 6 hours = 250 days till 0)
+        calc.file(bytes32("tau"), 21_600_000);
+
+        // Cusp not relevant, stays the same (tail will stop the auction)
+        // (Currently 90% decrease)
+
+        // Set tail to 432,000 second (5 days, implies minimum price of 0.98)
+        Clipper(clipper).file("tail", 432_000);
+        tail = Clipper(clipper).tail();
+
+        // Set hole on 5m
+        Dog(dog).file("TUSD-A", "hole", 5_000_000 * RAD);
+
+        // Set chip to 0 (no need to be fast)
+        Clipper(clipper).file("chip", 0);
+
+        // Set tip to 500 (kick all at start to achieve linear price discount, a bit higher value so that we guarantee kicking them all - ideally PE helps here)
+        Clipper(clipper).file("tip", 500 * RAD);
+
+        renounceOwnership(clipper);
+        renounceOwnership(dog);
+    }
+
+    function newAuction(uint256 amt) internal {
+        giveTokens(tusd, amt);
+
+        Token(tusd).approve(gemJoin, amt);
+        Join(gemJoin).join(address(this), amt);
+        (, uint256 rate, uint256 spot,,) = Vat(vat).ilks("TUSD-A");
+        uint256 maxArt = amt * spot / rate;
+
+        // Support creating more positions (for testing only)
+        takeOwnership(vat);
+        Vat(vat).file("TUSD-A", "line", type(uint256).max);
+        renounceOwnership(vat);
+
+        Vat(vat).frob({
+            i:    "TUSD-A",
+            u:    address(this),
+            v:    address(this),
+            w:    address(this),
+            dink: int(amt),
+            dart: int(maxArt)
+        });
+
+        // Set LR to 150%
+        takeOwnership(spotter);
+        Spot(spotter).file("TUSD-A", bytes32("mat"), 150 * RAY / 100);
+        renounceOwnership(spotter);
+        Spot(spotter).poke("TUSD-A");
+
+        id = Dog(dog).bark("TUSD-A", address(this), address(this));
+    }
+
+    function test_baseline() public {
+        uint256 amt = 20000 * WAD;
+        newAuction(amt);
+        bytes memory data = abi.encode(
+            address(123),
+            address(gemJoin),
+            uint256(0)
+        );
+        Hevm(hevm).warp(block.timestamp + tail / 2);
+        Clipper(clipper).take({
+            id:   id,
+            amt:  amt,
+            max:  type(uint256).max,
+            who:  address(callee),
+            data: data
+        });
+
+        assertEq(Token(dai).balanceOf(address(this)), 0);
+        assertEq(Token(tusd).balanceOf(address(this)), 0);
```

Comment:
I am not sure I understand these checks. I would suggest to check `clipper.count() == prevCount - 1` instead

Reply:
Added count checls.